mlir/lib/Conversion/MPIToLLVM/MPIImplTraits.h

```diff
@@ -0,0 +1,157 @@
+#define MPICH_SKIP_MPICXX 1
+#define OMPI_SKIP_MPICXX 1
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+#include "mlir/Dialect/MPI/IR/MPI.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+// skip if no MPI C header was found
+#ifdef FOUND_MPI_C_HEADER
+#include <mpi.h>
+#else // not FOUND_MPI_C_HEADER
+#include "mpi_fallback.h"
+#endif // FOUND_MPI_C_HEADER
```

Comment:
I'm not sure if this is acceptable, tbh. This should probably be a configuration option of the pass, not something taking from header files. This is especially problematic when building on one machine and then distributing the produced binaries to some other system which might now have these headers.

Reply:
Are you suggesting to make this "runtime" options, e.g. chose the right values/configuration when the compiler is running?
I agree, that would of course be ideal. I suspect that that it would overcomplicate things in the current state of affairs. At this point, the MPI support is very nascent, and MLIR is rarely (if ever) distributed in binary form. 
We can always improve in future PRs, in particular if the new ABI turns out to be insufficient or is taking too long to materialize.

Reply:
I am not aware of a tangible plan. Probably mostly because no concrete suggestion was made yet. 
I'd be happy to work with you on that, but I suggest doing this in a separate PR.

Notice: Even though not ideal, the current default/fallback approach works for all the MPICH compatible MPI implementations (https://www.mpich.org/abi/).